src/qr/dclogin_scheme.rs

```diff
@@ -0,0 +1,388 @@
+
+        // apply to result struct
+        let options: LoginOptions = match parameter_map.get("v").map(|i| i.parse::<u32>()) {
+            Some(Ok(1)) => LoginOptions::V1 {
+                mail_pw: parameter_map
+                    .get("p")
+                    .map(|s| s.to_owned())
+                    .context("password missing")?,
+                imap_host: parameter_map.get("ih").map(|s| s.to_owned()),
+                imap_port: parse_port(parameter_map.get("ip"))
+                    .context("could not parse imap port")?,
+                imap_username: parameter_map.get("iu").map(|s| s.to_owned()),
+                imap_password: parameter_map.get("ipw").map(|s| s.to_owned()),
+                imap_security: parse_socket_security(parameter_map.get("is"))?,
+                imap_certificate_checks: parse_certificate_checks(parameter_map.get("ic"))?,
+                smtp_host: parameter_map.get("sh").map(|s| s.to_owned()),
+                smtp_port: parse_port(parameter_map.get("sp"))
+                    .context("could not parse smtp port")?,
+                smtp_username: parameter_map.get("su").map(|s| s.to_owned()),
+                smtp_password: parameter_map.get("spw").map(|s| s.to_owned()),
+                smtp_security: parse_socket_security(parameter_map.get("ss"))?,
+                smtp_certificate_checks: parse_certificate_checks(parameter_map.get("sc"))?,
+            },
+            Some(Ok(v)) => LoginOptions::UnsuportedVersion(v),
+            Some(Err(_)) => bail!("version could not be parsed as number E6"),
+            None => bail!("invalid DCLOGIN payload: version missing E7"),
+        };
+
+        Ok(Qr::Login {
+            address: addr.to_owned(),
+            options,
+        })
+    } else {
+        bail!("Bad scheme for account URL: {:?}.", payload);
+    }
+}
+
+fn parse_port(port: Option<&String>) -> core::result::Result<Option<u16>, std::num::ParseIntError> {
+    match port {
+        Some(p) => Ok(Some(p.parse::<u16>()?)),
+        None => Ok(None),
+    }
+}
+
+fn parse_socket_security(security: Option<&String>) -> Result<Option<Socket>> {
+    Ok(match security.map(|s| s.as_str()) {
+        Some("ssl") => Some(Socket::Ssl),
+        Some("starttls") => Some(Socket::Starttls),
+        Some("default") => Some(Socket::Automatic),
+        Some("plain") => Some(Socket::Plain),
+        Some(other) => bail!("Unknown security level: {}", other),
+        None => None,
+    })
+}
+
+fn parse_certificate_checks(
+    certificate_checks: Option<&String>,
+) -> Result<Option<CertificateChecks>> {
+    Ok(match certificate_checks.map(|s| s.as_str()) {
+        Some("0") => Some(CertificateChecks::Automatic),
+        Some("1") => Some(CertificateChecks::Strict),
+        Some("3") => Some(CertificateChecks::AcceptInvalidCertificates),
+        Some(other) => bail!("Unknown certificatecheck level: {}", other),
+        None => None,
+    })
+}
+
+pub(crate) async fn apply_from_login_qr(
```

Comment:
nit, maybe call this `configure_from_login_qr`?

Reply:
i think, this is resolved?